include/hip/channel_descriptor.h

```diff
@@ -28,12 +28,22 @@ THE SOFTWARE.
 // on NVCC path:
 
 
-#if (defined(__HIP_PLATFORM_HCC__) || defined(__HIP_PLATFORM_AMD__)) && !(defined(__HIP_PLATFORM_NVCC__) || defined(__HIP_PLATFORM_NVIDIA__))
+#if (defined(__HIP_PLATFORM_HCC__) || defined(__HIP_PLATFORM_AMD__)) &&                            \
+    !(defined(__HIP_PLATFORM_NVCC__) || defined(__HIP_PLATFORM_NVIDIA__)) &&                       \
+    !(defined(__HIP_PLATFORM_CLANG__) || defined(__HIP_PLATFORM_SPIRV__))
 #include <hip/amd_detail/amd_channel_descriptor.h>
-#elif !(defined(__HIP_PLATFORM_HCC__) || defined(__HIP_PLATFORM_AMD__)) && (defined(__HIP_PLATFORM_NVCC__) || defined(__HIP_PLATFORM_NVIDIA__))
+
+#elif (defined(__HIP_PLATFORM_NVCC__) || defined(__HIP_PLATFORM_NVIDIA__)) &&                      \
+    !(defined(__HIP_PLATFORM_HCC__) || defined(__HIP_PLATFORM_AMD__)) &&                           \
+    !(defined(__HIP_PLATFORM_CLANG__) || defined(__HIP_PLATFORM_SPIRV__))
 #include <hip/nvidia_detail/nvidia_channel_descriptor.h>
-#else
-#error("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__");
-#endif
 
-#endif
+#elif (defined(__HIP_PLATFORM_CLANG__) || defined(__HIP_PLATFORM_SPIRV__)) &&                      \
+    !(defined(__HIP_PLATFORM_HCC__) || defined(__HIP_PLATFORM_AMD__)) &&                           \
+    !(defined(__HIP_PLATFORM_NVCC__) || defined(__HIP_PLATFORM_NVIDIA__))
+#include <hip/spirv_channel_descriptor.h>
```

Comment:
For HIP/AMD and HIP/NVCC, the header files containing implementation details are in a separate repo [hipamd](https://github.com/ROCm-Developer-Tools/hipamd). However, the installed header files form a unified directory structure:
include - hip+
                     |-amd_detail
                     |-nvidia_detail
 where include/hip contains generic HIP headers, include/hip/amd_detail contains AMD/HIP specific headers and include/hip/nvidia_detail contains NV/HIP specific headers.

If more platforms are supported, it is desirable to keep this directory structure consistent and clean.

Will SPIRV specific HIP headers be upstreamed to https://github.com/ROCm-Developer-Tools/hipamd ? If not, where will they be? Is it possible to keep them in include/hip/spirv_detail to keep the directory structure clean?

Reply:
> Will SPIRV specific HIP headers be upstreamed to https://github.com/ROCm-Developer-Tools/hipamd ? 

no plans for that. 

> In the current repo much like hipamd

>  Is it possible to keep them in include/hip/spirv_detail to keep the directory structure clean?

ok we can rename the directories as you requested.